jscomp/core/js_record_iter.ml

```diff
@@ -0,0 +1,149 @@
+
+    open J  
+    let unknown _ _ = ()
+    let option sub self = fun v -> 
+      match v with 
+      | None -> ()
+      | Some v -> sub self v
+    let rec list sub self = fun x  -> 
+      match x with 
+      | [] -> ()
+      | x::xs -> 
+         sub self x ;
+        list sub self xs
+
+    type iter = {
+      label : label fn;
+required_modules : required_modules fn;
+ident : ident fn;
+module_id : module_id fn;
+vident : vident fn;
+exception_ident : exception_ident fn;
+for_ident : for_ident fn;
+for_direction : for_direction fn;
+property_map : property_map fn;
+length_object : length_object fn;
+expression_desc : expression_desc fn;
+for_ident_expression : for_ident_expression fn;
+finish_ident_expression : finish_ident_expression fn;
+statement_desc : statement_desc fn;
+expression : expression fn;
+statement : statement fn;
+variable_declaration : variable_declaration fn;
+string_clause : string_clause fn;
+int_clause : int_clause fn;
+case_clause : case_clause fn;
+block : block fn;
+program : program fn;
+deps_program : deps_program fn
+    }  
+    and 'a fn = iter -> 'a -> unit
+    let iter : iter = {
+     label : label fn  = ( unknown )  ;
+ required_modules : required_modules fn  = ( fun _self arg -> list ((fun _self arg -> _self.module_id _self arg)) _self arg )  ;
+ ident : ident fn  = ( unknown )  ;
+ module_id : module_id fn  = ( fun _self { id = _x0;kind = _x1} -> begin _self.ident _self _x0 end )  ;
+ vident : vident fn  = ( fun _self -> function 
+| Id ( _x0)  -> 
+ begin _self.ident _self _x0 end
+|Qualified ( _x0,_x1)  -> 
+ begin _self.module_id _self _x0 end )  ;
+ exception_ident : exception_ident fn  = ( (fun _self arg -> _self.ident _self arg) )  ;
+ for_ident : for_ident fn  = ( (fun _self arg -> _self.ident _self arg) )  ;
+ for_direction : for_direction fn  = ( unknown )  ;
+ property_map : property_map fn  = ( fun _self arg -> list ((fun _self (_x0,_x1) -> begin _self.expression _self _x1 end)) _self arg )  ;
```

Comment:
this is actually hard to simplify -- we can create a closure using `_self`, but not sure if it is worth it